### PR TITLE
Implement login feature for users and UI improvements

### PR DESCRIPTION
**Pull request by tonypan2000**
1. post comment form stays hidden until user logs in
2. user can choose to display a nickname or their login email (default)
3. user can only delete their own comment

#6

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 1]
```diff
@@ -17,10 +19,26 @@
...
     Key commentKey = KeyFactory.createKey("Comment", id);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(commentKey);
+
+    String posterEmail = "";
+    String userEmail = userService.getCurrentUser().getEmail();
+    try {
+      posterEmail = (String) datastore.get(commentKey).getProperty("email");
+    } catch (Exception e) {
+      System.err.println(e);
```
Comment (ahorjia) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:34]: I am curious to know where this log is written to? Is there to have a permanent log for your web application. This looks like it is writing to a console but where is that concole?

Comment (ahorjia) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:34]: Maybe we should return an error to the user?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -40,17 +42,17 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  private class Comment {
+  private static class Comment {
```
[the remark this answers is not in the document]
Comment (tonypan2000) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: It was suggested by Jess in #4, and it showed up again due to me merging another branch I was working on. The static class Comment cannot access any private instances of the Dataservlet class.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 1 above)
Comment (tonypan2000) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:34]: Yes, I need to return the error to user.

**Comment by jessrenteria** [comment]
Whoops. Didn't see you already merged. Please take a look at the comments I left and consider following up in another PR.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -112,36 +114,41 @@ private int getMaxNumComments(HttpServletRequest request, int input) {
...
-
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("name", name);
-    commentEntity.setProperty("timestamp", currentTimeEpoch);
-    commentEntity.setProperty("content", comment);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
```
[the remark this answers is not in the document]
Comment (tonypan2000) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:118]: I already check whether user is logged in on the client side and sends an alert, so unlogged in users should never be able to get inside this function call. Maybe I will remove this if statement

**Pull request merged**